sql/src/test/java/io/crate/expression/scalar/string/StringLeftRightFunctionTest.java

```diff
@@ -0,0 +1,123 @@
+        assertEquals(StringLeftRightFunction.right(str, -10 * str.length()), "");
+        assertEquals(StringLeftRightFunction.right(null, 10), "");
+    }
+
+    @Test
+    public void testNormalizeLeftFunc() throws Exception {
+        assertNormalize("left('crate.io', 0)", isLiteral(""));
+        assertNormalize("left('crate.io', 5)", isLiteral("crate"));
+        assertNormalize("left('crate.io', 8)", isLiteral("crate.io"));
+        assertNormalize("left('crate.io', 100)", isLiteral("crate.io"));
+        assertNormalize("left('crate.io', -3)", isLiteral("crate"));
+        assertNormalize("left('crate.io', -8)", isLiteral(""));
+        assertNormalize("left('crate.io', -100)", isLiteral(""));
+        assertNormalize("left(3, 1)", isLiteral("3")); // casted to text
+        assertNormalize("left(3, -1)", isLiteral(""));
+        assertNormalize("left(null, -1)", isLiteral(null));
+    }
+
+    @Test
+    public void testEvaluateLeftFunc() throws Exception {
+        Literal<Object> value = Literal.of(DataTypes.STRING, "crate.io");
+        assertEvaluate("left('crate.io', 0)", "", value);
+        assertEvaluate("left(name, 5)", "crate", value);
+        assertEvaluate("left(name, 8)", "crate.io", value);
+        assertEvaluate("left(name, 100)", "crate.io", value);
+        assertEvaluate("left(name, -3)", "crate", value);
+        assertEvaluate("left(name, -8)", "", value);
+        assertEvaluate("left(name, -100)", "", value);
+        assertEvaluate("left(name, -1)", null, NULL_LITERAL);
+    }
+
+    @Test
+    public void testNormalizeRightFunc() throws Exception {
+        assertNormalize("right('crate.io', 0)", isLiteral(""));
+        assertNormalize("right('crate.io', 2)", isLiteral("io"));
+        assertNormalize("right('crate.io', 8)", isLiteral("crate.io"));
+        assertNormalize("right('crate.io', 100)", isLiteral("crate.io"));
+        assertNormalize("right('crate.io', -6)", isLiteral("io"));
+        assertNormalize("right('crate.io', -8)", isLiteral(""));
+        assertNormalize("right('crate.io', -100)", isLiteral(""));
+        assertNormalize("right(3, 1)", isLiteral("3"));
+        assertNormalize("right(3, -1)", isLiteral(""));
+        assertNormalize("right(null, -1)", isLiteral(null));
+    }
+
+    @Test
+    public void testEvaluateRightFunc() throws Exception {
+        Literal<Object> value = Literal.of(DataTypes.STRING, "crate.io");
+        assertEvaluate("right('crate.io', 0)","", value);
+        assertEvaluate("right('crate.io', 2)","io", value);
+        assertEvaluate("right('crate.io', 8)","crate.io", value);
+        assertEvaluate("right('crate.io', 100)","crate.io", value);
+        assertEvaluate("right('crate.io', -6)","io", value);
+        assertEvaluate("right('crate.io', -8)","", value);
+        assertEvaluate("right('crate.io', -100)","", value);
+        assertEvaluate("right(null, -1)",null, NULL_LITERAL);
+    }
```

Comment:
Most of the test cases here are more testing the java `String.substring()` function than our scalar function code.
Would be great to minimize the tests to what is really needed.

Reply:
Ok!, will update.